### ISSUE TITLE
What does this proposed API provide?

### ISSUE THREAD

**Issue by BenLubar** [RupertBenWiser/Web-Environment-Integrity#31]
I read the proposal and it's not clear to me how this provides anything that the fetch or XMLHttpRequest APIs don't already provide, unless there's some undocumented user information that's being transmitted to the attester. It's also not clear how the attester's key becomes trusted. I see three options:

1. There are user agent-provided attesters, which means this proposal *forces* vendor lock-in because each vendor's public key will need to be individually trusted.
2. There are website-provided attesters, which means this proposal offers no features that XMLHttpRequest or fetch can't already do, unless there's some form of user data being transmitted to the attester (again, the spec is very vague about what gets sent).
3. Attesters can be configured by the user, which means this is an API to get an arbitrary chunk of usually meaningless data.

- [ ] Document what information is sent to the attester and how the attester verifies that information is "correct" before signing a payload
- [ ] If this information is sensitive, document why it's okay to send it to the attester when it wouldn't be okay to expose it to the website directly
- [ ] Document how a malicious actor would be stopped by this proposed security feature

**Comment by mokrates** [comment]
The site could send a cookie to the Chrome, Chrome is installed via Playstore on your unlocked phone, Google Services on your Phone check, that the APK-Checksum is signed correctly, signal Google. Your Chrome gets the cookie signed by Google and you can send it back to the website. If the signature doesn't include that it is by google, but only that it is "ok" somehow (idk... Main CA or sth...), then that would leak possibly only little information.

Nonetheless, the restrictions that come to opensource software and the user are not in any way ok. Just having the idea is reeeeeal bad.

**Comment by BenLubar** [comment]
> The site could send a cookie to the Chrome, Chrome is installed via Playstore on your bootlocked phone, Google Services on your Phone check, that the APK-Checksum is signed correctly, signal Google.

Certainly, but the same thing could be done by setting up a loopback web server on a static port on every Android phone that responds to CORS requests for a signed payload, and it would require exactly the same amount of work and information disclosure.

Additionally, RFC9053 signatures do not have any time information included. RFC8949 objects can include timestamps, but there is nowhere in this specification that defines what the actual payload contains.

I would like to add an additional question to the above:

- [ ] Can the data in the payload be used to identify a user? If so, this API essentially just provides the ability to fingerprint users. If not, what's stopping the payload from a different user being sent? This API does not require any permissions, so a malicious script could easily harvest millions of these identifiers undetected.

**Comment by mokrates** [comment]
Why would a timestamp be relevant?

**Comment by BenLubar** [comment]
What's to stop me from returning the same payload every time the function is called? The specification simply says that you can verify the signature and doesn't say anything about the content of the payload other than what encoding format it uses.

**Comment by mokrates** [comment]
look you here.
https://github.com/RupertBenWiser/Web-Environment-Integrity/issues/31#issuecomment-1642659966

random challenge token. generated with a distributed algo between server and client. so the server can't include a tracking-cookie, there is no replay-attack, etc.

**Comment by mokrates** [comment]
Yes, the specs probably don't say these things. And I doubt that this can be in favour of the user in any way. But it's possible to implement it such that it leaks very little information.

The problem is the lock out of other browsers and plugins.

**Comment by BenLubar** [comment]
Here is the entirety of the specification for the object returned by the attester:

https://rupertbenwiser.github.io/Web-Environment-Integrity/#attester-verdict

> The attester verdict consists of:
> - A raw response (a COSE [RFC9053] signed CBOR [RFC8949] stored in an ArrayBuffer

That's it. That's the entire documentation.

There's no mention that the encoded object contains *anything*, so a conforming implementation could (and should, for optimal performance) return a statically allocated buffer containing a digitally signed empty CBOR object.

**Comment by mokrates** [comment]
It says it's a CBOR and the spec isn't done yet. The CBOR will contain sth, that is signed. And has to. A timestamp probably won't help. You will have to sign some token. The question is, what does the token contain, and how to ensure, it doesn't, but still can do it's job.

An empty Message with no timestamp won't do, and an empty Message WITH a timestamp probably won't do either (b/c it would have a ttl)
